src/js/components/alert.js

```diff
@@ -5,43 +5,67 @@
 
 export default class Alert {
 
-  constructor(options) {
-    this.handleClick = this.handleClick.bind(this);
+  constructor(element) {
 
-    this.init(options);
-  }
-
-  init(options) {
-    document.getElementById(options.id).closest('[aria-labelledby]').querySelector('[data-alert-close]').addEventListener('click', () => { this.handleClick(options.id); }, false);
-  }
+    // Instance properties
+    this.element = element;
+    this.closeAttribute = ['data-alert-close'];
+    this.closeSelector = `[${this.closeAttribute}]`;
+    this.ariaAttribute = ['aria-labelledby'];
+    this.ariaSelector = `[${this.ariaAttribute}]`;
 
-  handleClick(id) {
-    const dismissButton = event.target.closest('[data-alert-close]');
+    // bind methods
+    this._handleClick = this._handleClick.bind(this);
 
-    // If the target wasn't the dismiss button bail.
-    if (!dismissButton) return;
+    // Check to make sure that a DOM element was passed in for initialization
+    if (!(Array.isArray(this.element))) {
+      throw new TypeError(
+        'A DOM element should be passed as the first argument to initialize the /alert'
+      );
+    }
 
-    this.dismissAlert(id);
+    this.init();
   }
 
-  dismissAlert(id, callback) {
-    let alert = document.querySelector('[aria-labelledby="' + id + '"]');
+  _handleClick(event) {
+    const dismissAria = event.target.closest(this.ariaSelector);
 
-    if (!alert) {
-      alert = document.getElementById(id);
+    // If the target wasn't the Aria element bail.
+    if (!dismissAria) {
+      throw new Error(
+        'Couldn\'t locate the Aria selector for the element.'
+      );
     }
 
+    this.dismissAlert(dismissAria);
+
+  }
+
+  dismissAlert(dismissAria) {
+    let alert = dismissAria;
+
     if (!alert) {
       throw new Error(
-        'Could not find an alert with the id of ' + id + ' to dismiss.'
+        'Could not find an alert attached to the close selector to dismiss.'
       );
     }
 
     alert.remove();
+  }
 
-    if (callback && typeof callback === 'function') {
-      callback();
-    }
+  init() {
+    // Add click handlers
+    this.element.forEach((item) => {
+      let dismissButton = item.querySelector(this.closeSelector);
+      dismissButton.addEventListener('click', this._handleClick, false);
+    })
+  }
+
+  destroy() {
```

Comment:
You'll need to mimic the set up in the `init()` method when you refactor this based on the comments above.